src/pages/ReflectionsPage/shared/ReflectionForm.js

```diff
@@ -66,16 +66,22 @@ const ReflectionForm = ({ reflection }) => {
 
   const handleSubmitClick = async () => {
     const answerDocs = answers
-      .filter((answer, index) => {
-        const question = questions[index];
-        return question.choices?.find(({ body }) => body === answer) != null;
-      })
       .map((answer, index) => {
         const question = questions[index];
```

Comment:
There was a subtle bug here: `const question = questions[index]` can possibly be incorrect in the map after the filter, because filtering out elements changes the number of elements and changes the index

Reply:
good catch!